x-pack/legacy/plugins/file_upload/public/components/index_settings.js

```diff
@@ -8,19 +8,30 @@ import React, { Fragment, Component } from 'react';
 import { i18n } from '@kbn/i18n';
 import { EuiFormRow, EuiFieldText, EuiSelect, EuiCallOut } from '@elastic/eui';
 import { FormattedMessage } from '@kbn/i18n/react';
-import { getExistingIndices, getExistingIndexPatterns }
-  from '../util/indexing_service';
+import {
+  getExistingIndexNames,
+  getExistingIndexPatternNames,
+  checkIndexPatternValid,
+} from '../util/indexing_service';
 
 export class IndexSettings extends Component {
 
   state = {
     indexNameError: '',
     indexDisabled: true,
-    indexPatterns: null,
-    indexNames: null,
     indexName: '',
+    indexNameList: [],
+    indexPatternList: [],
   };
 
+  async componentDidMount() {
+    this.setState({
+      // Set once on component load
```

Comment:
comment is redundant

Reply:
asynciness can cause errors.

await for the responses, but do not call `setState` if the component is no longer mounted. There's examples in the Maps-app, e.g. https://github.com/thomasneirynck/kibana/blob/e779ae71be136352bd4e0355d2f65af40051ce46/x-pack/legacy/plugins/maps/public/components/layer_panel/view.js#L84